hbase-plugins/src/main/java/co/cask/hydrator/plugin/sink/mapreduce/HBaseTableOutputFormat.java

```diff
@@ -27,16 +36,78 @@
  */
 public class HBaseTableOutputFormat<KEY> extends TableOutputFormat<KEY> {
 
+  private static final Logger LOG = LoggerFactory.getLogger(HBaseTableOutputFormat.class);
+
+  public static final String HBASE_CUSTOM_TABLENAME = "hbase.custom.tablename";
+  public static final String HBASE_CUSTOM_COLUMNFAMILY = "hbase.custom.Columnfamily";
+
   @Override
   public void setConf(Configuration otherConf) {
     // To resolve CDAP-12731, set the current class's classloader to the thread's context classloader,
     // so that it gets picked up when the super.setConf() calls HBaseConfiguration.create(Configuration)
     ClassLoader originalClassLoader = Thread.currentThread().getContextClassLoader();
     Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
+    HBaseAdmin admin = null;
     try {
       super.setConf(otherConf);
+      Configuration configuration = super.getConf();
+      admin = new HBaseAdmin(configuration);
+      createOrUpdateTable(configuration, admin);
+    } catch (IOException e) {
+      LOG.error("Error while creating hbase table ", e);
+      throw new RuntimeException(e);
```

Comment:
why we are converting it into RunTime ? let it be original exception

Reply:
we can't throw directly IOExcption as it is checked exception.